client_test.go

```diff
@@ -1735,3 +1735,50 @@ func startEchoServerExt(t *testing.T, network, addr string, isTLS bool) *testEch
 		t:  t,
 	}
 }
+
+func newLocalListener(t testing.TB) net.Listener {
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		ln, err = net.Listen("tcp6", "[::1]:0")
```

Comment:
Is there a specific reason why you added `tcp6` as fallback here? We don't do this anywhere else. It should either be done everywhere or nowhere.

Reply:
Done

Reply:
It's a code from golang tls package